tests/daysOfTheWeek.js

```diff
@@ -17,3 +18,15 @@ tap.test('getDayName', (getDayNameTest) => {
   getDayNameTest.equal(target.getDayName(validDateObject), 'Friday', 'get expected day name when using a Date object');
   getDayNameTest.end();
 });
+
+tap.test('next function', (nextTests) => {
+  const testDate = new Date(2017, 5, 23);
+  mockdate.set(testDate);
+  const expectedDate = new Date(2017, 5, 25);
+
+  nextTests.throws(() => target.next('January'), 'throws when input is not a day of the week');
+  nextTests.equal(target.next('Sunday').getTime(), expectedDate.getTime(), 'get expected date for next Sunday');
```

Comment:
Hmm!  Looking at the code in the `next` function, I can see that there's some conditional behavior.  It depends on whether today is before or after the request day of the week (that is, if today is Friday and we ask for the next Tuesday, the function will behave slightly different than if I ask for the next Saturday).  However, the tests don't guarantee that we will always check that condition.  We should make sure that tests cover all of the possible conditional behaviors.  To be sure this one is covered, we should add a test for a day of the week *after* the test date.

### Neat note
This test makes use of a "mock."  A mock is a way of faking some piece of software that impacts the code we're trying to test but isn't in that code itself.  In this case, the current date will affect how the tests run, so these tests "mock" the date - they make the code think "today" is always June 23, 2017.  That way we can know with absolute certainty what the outputs ***should*** be.